### ISSUE TITLE
[11] odoo field state using in attribs must be in view but it is missing in Purchase_order_secondary_unit

### ISSUE THREAD

**Issue by karlggest** [OCA/purchase-workflow#701]
Hi.

I try to install purchase_order_secondary_unit it fails with the message:
`field state using in attribs must be in view but it is missing in`

If I add the line 
`<field name="state" invisible="1"/>`
in the first xpath (line 11) it seems to works.

It's the right arrangement to this?

Thanks and regards.

**Comment by pedrobaeza** [comment]
Is this happening in runbot?

**Comment by karlggest** [comment]
> Is this happening in runbot?

Hi.

I think I can't even using that: http://runbot.odoo.com/runbot

I test this module downloading the entire repo (as product-attribute and sale-workflow too) in a test odoo install (CentOS, odoo 11, python 3, venv). I have a few databases, and try to install this module. One of my databases showed an error but module works (but no control of stock, sales shows "no stock" even if in report->inventory I can see products stock). I then try others databases and always failed.

Searching internet I find a similar error text (field type, I think) and the suggestion try to add the line with invisible="1", and this seems work. Only I can see is that dialog "purchase secondary unit" in product form is visible even if I don't mark this option to my user.

Sorry for my awful english!

Thanks.

**Comment by pedrobaeza** [comment]
I meant OCA runbot:

![Selección_038](https://user-images.githubusercontent.com/7165771/59628043-f885db80-913f-11e9-8f91-d6b4e8db0d8e.png)

https://runbot.odoo-community.org/runbot/142/11.0

**Comment by karlggest** [comment]
Hello!
> I meant OCA runbot:
> 
> ![Selección_038](https://user-images.githubusercontent.com/7165771/59628043-f885db80-913f-11e9-8f91-d6b4e8db0d8e.png)
> 
> https://runbot.odoo-community.org/runbot/142/11.0
As I understand this, the system has the module already installed. I try it, see orders with secondary unit, all ok.

In my system works too with the workaround I wrote. But fails without it, except in one database works 

Thanks for your help.

**Comment by pedrobaeza** [comment]
But the problem is due to a module you have then, not by this module itself. Please check which one is conflicting.

**Comment by karlggest** [comment]
Well, If you mean an odoo official module, I don't know. I start a blank database with sales, puchases and inventory modules, add secondary unit modules, and it fails. I can't see a difference between sales_secondary_unit and purchases_secondary_unit. Really, where it works without changes is in the most different (a few different test tools, even POS). In that database it works fine and doesn't show warnings or errors. But in the most simple, it fails only in Purchases_secondary_unit.

Thanks for your help.

**Comment by pedrobaeza** [comment]
So you mean that installing an additional module, it doesn't fail? But it does on a fresh one?

**Comment by karlggest** [comment]
Hello.

I tried this module (and sales_secondary_unit and product_secondary_unit) in 4 databases:

* One test database similar to my production system. I have some OCA modules and a few special modules about POS. I installed the "secondary modules" and it works.
* Two test databases, mainly OCA modules, I use to my tests and learn. Both fails when I try to install purchase_secondary_unit, but sales_secondary_unit and product_secondary_unit works.
* One blank database. I installed the main modules I need to test (purchases, sales, inventory) and try the "secondary" modules. Same error than the two test databases.

In all cases, with the workaround I talked before, all works fine. But I can see "purchase secondary unit" box selection in the product form even before I select the right permission in the user form.

Thanks for your help :)
